### ISSUE TITLE
AOT : "Supplied parameters do not match any signature of call target"

### ISSUE THREAD

**Issue by ThomasKientz** [swimlane/ngx-charts#274]
**I'm submitting a ...** 
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
When building with AOT, I got the error :

> [at-loader] Checking finished with 26 errors
Error in bail mode: [at-loader] compiled/node_modules/@swimlane/ngx-charts/release/bar-chart/bar-horizontal-2d.component.ngfactory.ts:784:35 
    TS2346: Supplied parameters do not match any signature of call target.

**Please tell us about your environment:**
I'm using [the AngularClass webpack seed](https://github.com/AngularClass/angular2-webpack-starter)

* **ngx-charts version:** 4.2.1

* **Angular version:** 2.4.6

<img width="682" alt="capture d ecran 2017-03-11 a 16 12 38" src="https://cloud.githubusercontent.com/assets/10157845/23822609/97cb6398-0675-11e7-856e-0e214953659d.png">

**Comment by botmonster** [comment]
I confirm this with following versions:

- **ngx-charts version:** 4.3.0
- **Angular version:** 2.4.9

**Full error output:**

```
ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-2d.component.ngfactory.ts:784:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-2d.component.ngfactory.ts:788:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-normalized.component.ngfactory.ts:746:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-normalized.component.ngfactory.ts:750:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-stacked.component.ngfactory.ts:746:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-horizontal-stacked.component.ngfactory.ts:750:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-2d.component.ngfactory.ts:782:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-2d.component.ngfactory.ts:786:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-normalized.component.ngfactory.ts:746:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-normalized.component.ngfactory.ts:750:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-stacked.component.ngfactory.ts:746:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bar-chart\bar-vertical-stacked.component.ngfactory.ts:750:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\bubble-chart\bubble-chart.component.ngfactory.ts:696:41
    TS2339: Property 'clipPathId' does not exist on type 'BubbleChartComponent'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\force-directed-graph\force-directed-graph.component.ngfactory.ts:207:35
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\gauge\gauge-axis.component.ngfactory.ts:131:7
    TS2346: Supplied parameters do not match any signature of call target.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\gauge\gauge-axis.component.ngfactory.ts:373:50
    TS2339: Property 'big' does not exist on type 'any[]'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\gauge\gauge-axis.component.ngfactory.ts:376:50
    TS2339: Property 'big' does not exist on type 'any[]'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\gauge\gauge-axis.component.ngfactory.ts:379:50
    TS2339: Property 'small' does not exist on type 'any[]'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\heat-map\heat-map.component.ngfactory.ts:642:48
    TS2339: Property 'onActivate' does not exist on type 'HeatMapComponent'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\heat-map\heat-map.component.ngfactory.ts:646:48
    TS2339: Property 'onDeactivate' does not exist on type 'HeatMapComponent'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\pie-chart\advanced-pie-chart.component.ngfactory.ts:342:44
    TS2339: Property 'labels' does not exist on type 'AdvancedPieChartComponent'.

ERROR in [at-loader] compiled\node_modules\@swimlane\ngx-charts\release\pie-chart\pie-label.component.ngfactory.ts:300:67
    TS2346: Supplied parameters do not match any signature of call target. ```

**Comment by CaerusKaru** [comment]
Will this be resolved soon in a minor/patch release? Getting the same thing still using angular-cli with AoT (which no longer masks the warnings)

**Comment by DarthFarzad** [comment]
I managed to resolve this by uninstalling angular-cli ,clearing the cache and reinstalling.

**Comment by omeralper** [comment]
Uninstalling and reinstalling angular-cli didn't help me. I get "Supplied parameters do not match any signature of call target. " error and since build fails I cannot see produced ngfactory file.

**Comment by marjan-georgiev** [comment]
This should be fixed in 5.2.1. Please let me know if it's still occurring.

**Comment by jboothe** [comment]
When will it be fixed in the 4.x branch? (or how would one go about fixing it?) We've made a commitment to Angular 2 and cannot upgrade ngx-charts to 5.x.
